### PR TITLE
Filters totally rebuilded.

### PR DESCRIPTION
**Pull request by NikKovIos**
The features:

- Added a support for the custom filters, not only CoreImage;
- Added a few instagram filters (taken from https://github.com/i05nagai/ios-instagram-like-filter);
- Rebuilded the thumbnail creating, that it happens at a moment and not freeze UI when the big image is selected;

### CODE REVIEW

File: Resources/id.lproj/Localizable.strings [hunk 1]
```diff
@@ -19,4 +19,4 @@
 "YPImagePickerVideoDurationTitle" = "Durasi video";
 "YPImagePickerVideoTooLong" = "Pilih durasi video kurang dari %@ detik";
 "YPImagePickerVideoTooShort" = " Durasi video setidaknya %@ detik";
-"YPImagePickerWarningItemsLimit" = "Hanya diperbolehkan %d foto atau video";
```
Comment (s4cha) [inline Resources/id.lproj/Localizable.strings:22]: Let's try not to include unrelated changes, I know Xcode loves to change things without us knowing ;)

File: Source/Filters/YPPhotoFiltersVC.swift [hunk 2]
```diff
@@ -118,48 +115,68 @@ open class YPPhotoFiltersVC: UIViewController, IsMediaFilterVC, UIGestureRecogni
...
+            let ciImage = inputPhoto.originalImage.toCIImage(),
+            let modifiedFullSizeImage = applier(ciImage) {
+            inputPhoto.modifiedImage = modifiedFullSizeImage.toUIImage()
+        } else {
+            inputPhoto.modifiedImage = nil
         }
         didSave?(YPMediaItem.photo(p: inputPhoto))
     }
+    
+    fileprivate func thumbFromImage(_ img: UIImage) -> CIImage {
```
Comment (s4cha) [inline Source/Filters/YPPhotoFiltersVC.swift:129]: I saw that the thumbnail image improved, this is very cool :)

File: Source/Filters/YPPhotoFiltersVC.swift [hunk 3]
```diff
@@ -118,48 +115,68 @@ open class YPPhotoFiltersVC: UIViewController, IsMediaFilterVC, UIGestureRecogni
     
     @objc
     func save() {
```
Comment (s4cha) [inline Source/Filters/YPPhotoFiltersVC.swift:117]: Only processing the full image at the end on save is also a great optimization 👏

**Review by s4cha — COMMENTED** [review]
👌

**Comment by s4cha** [comment]
- Opening Filters to custom ones is a great addition 👍 

- Concerning the UI freeze, I believe this was only slow on simulator, but not on device but anyway this looks faster :)

- I don't know if the filters taken from  https://github.com/i05nagai/ios-instagram-like-filter have been only inspired or copy pasted but I have a concern concerning the license. Indeed the repo doesn't have one.
 
From "https://choosealicense.com/no-permission/"
"If you find software that doesn’t have a license, that generally means you have no permission from the creators of the software to use, modify, or share the software. Although a code host such as GitHub may allow you to view and fork the code, this does not imply that you are permitted to use, modify, or share the software for any purpose.

Your options:

Ask the maintainers nicely to add a license. Unless the software includes strong indications to the contrary, lack of a license is probably an oversight. If the software is hosted on a site like GitHub, open an issue requesting a license and include a link to this site, or if you’re bold and it’s fairly obvious what license is most appropriate, open a pull request to add a license."

Could you contact @i05nagai to ask him to add a License?

File: Source/Filters/YPPhotoFiltersVC.swift (hunk 3 above)
Comment (NikKovIos) [reply Source/Filters/YPPhotoFiltersVC.swift:117]: Thanks)

File: Resources/id.lproj/Localizable.strings (hunk 1 above)
Comment (NikKovIos) [reply Resources/id.lproj/Localizable.strings:22]: This is Localize.swift made this changes. I think it would do it all the time, so better to commit once and not pay attention on this in future.

**Comment by NikKovIos** [comment]
I've created an issue https://github.com/i05nagai/ios-instagram-like-filter/issues/1

File: Resources/id.lproj/Localizable.strings (hunk 1 above)
Comment (s4cha) [reply Resources/id.lproj/Localizable.strings:22]: 👍

**Comment by s4cha** [comment]
I guess we can merge whenever we have the go for the License :)

**Comment by s4cha** [comment]
🚀
